src/alexandria3k/data_sources/issn_subject_codes.py

```diff
@@ -0,0 +1,218 @@
+#
+# Alexandria3k Crossref bibliographic metadata processing
+# Copyright (C) 2024  Panagiotis-Alexios Spanakis
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""ISSN Subject Codes data processing
+
+Requests per second allowed: 9 (https://dev.elsevier.com/api_key_settings.html)
+Total requests allowed: 25000 per week (https://dev.elsevier.com/api_key_settings.html)
+"""
+
+
+import csv
+import os
+import sqlite3
+import pybliometrics
+from pybliometrics.scopus import SerialSearch
+from alexandria3k.csv_source import CsvCursor, VTSource
+from alexandria3k.data_source import DataSource
+from alexandria3k.db_schema import ColumnMeta, TableMeta
+from alexandria3k import perf, debug
+from alexandria3k.common import ensure_table_exists, get_string_resource
+from alexandria3k.data_source import PROGRESS_BAR_LENGTH
+
+issn_subject_codes_table = TableMeta(
+    "issn_subject_codes",
+    cursor_class=CsvCursor,
+    delimiter=",",
+    columns=[
+        ColumnMeta("id", rowid=True),
+        ColumnMeta("issn"),
+        ColumnMeta("subject_code", data_type="INTEGER"),
+    ],
+)
+
+tables = [
+    issn_subject_codes_table,
+]
+
+
+class IssnSubjectCodes(DataSource):
+    """
+    Create an object containing ISSN to ASJC meta-data that supports queries over
+    its (virtual) table and the population of an SQLite database with its
+    data.
+
+    :param data_source: The location (file path or URL) where the data
```

Comment:
Only file path, not URL, right?
Also, "data are to be stored (for download) or loaded from (for populate or query)"

Reply:
Yes correct.